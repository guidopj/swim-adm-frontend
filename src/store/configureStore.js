import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import redirectMiddleware from 'middlewares/redirectMiddleware'
import history from 'history.js'
import { routerMiddleware } from 'connected-react-router'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

import rootSaga from 'sagas';
import rootReducer from 'reducers/rootReducer';

const persistConfig = {
	key: 'root',
	storage,
  }

  const mainReducer = rootReducer(history)
  const persistedReducer = persistReducer(persistConfig, mainReducer)

/** Saga Middleware */
const sagaMiddleware = createSagaMiddleware();

let middlewares = applyMiddleware(sagaMiddleware, routerMiddleware(history), redirectMiddleware);


export default function configureStore (initialState) {
	const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(persistedReducer,initialState, composeEnhancers(
        middlewares,
	  )
	);
	let persistor = persistStore(store)
	
	sagaMiddleware.run(rootSaga);
    return { store, persistor }
}

/* export const sagaMiddleware = createSagaMiddleware()

const composeStore = compose(
	applyMiddleware(
		redirectMiddleware,
		metricsMiddleware,
		sagaMiddleware,
		generalMiddleware,
		postMessageMiddleware
	),
	window.devToolsExtension ? window.devToolsExtension() : f => f
)(createStore)

const store = composeStore(reducers)

sagaMiddleware.run(rootSaga)

export default store */