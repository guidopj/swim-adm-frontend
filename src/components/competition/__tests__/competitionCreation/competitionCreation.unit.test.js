import React from 'react';
import CompetitionCreation from 'components/competition/creation/competitionCreation';
import renderer from 'react-test-renderer';
import history from 'history.js'
import { ConnectedRouter } from 'connected-react-router'
import { Provider } from 'react-redux'
import configureStore from 'store/configureStore'


describe ('Competition Creation with state and props', () => {

const { store } = configureStore({
    competitions: {}
  })

  it('renders correctly', () => {
    const component = renderer.create(
        <Provider store={store}>
            <ConnectedRouter history={history}>
                <CompetitionCreation
                />
            </ConnectedRouter>
        </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})