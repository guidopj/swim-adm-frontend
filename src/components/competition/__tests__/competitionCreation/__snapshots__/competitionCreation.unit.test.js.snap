// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competition Creation with state and props renders correctly 1`] = `
<div
  className="makeStyles-root-2"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-generalCard-6 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          New Competition: undefined
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink={false}
              htmlFor="club_name"
              id="club_name-label"
            >
              Club Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="club_name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-234 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-236"
                >
                  <span>
                    Club Name
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="pool_meters"
              id="pool_meters-label"
            >
              Select
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="pool_meters-label pool_meters"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="pool_meters"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                25
              </div>
              <input
                aria-describedby="pool_meters-helper-text"
                autoFocus={false}
                name="pool_meters"
                onAnimationStart={[Function]}
                type="hidden"
                value={25}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              className="MuiFormHelperText-root MuiFormHelperText-filled"
              id="pool_meters-helper-text"
            >
              Please select the pool meters
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-describedby="competition_start_date-helper-text"
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="competition_start_date"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <p
              className="MuiFormHelperText-root Mui-error"
              id="competition_start_date-helper-text"
            >
              Select a valid competition start date
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-describedby="competition_end_date-helper-text"
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="competition_end_date"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <p
              className="MuiFormHelperText-root Mui-error"
              id="competition_end_date-helper-text"
            >
              Select a valid competition end date
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-describedby="time-picker-helper-text"
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="time-picker"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change time"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <p
              className="MuiFormHelperText-root Mui-error"
              id="time-picker-helper-text"
            >
              Select a valid starting time
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-describedby="inscription_start_date-helper-text"
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="inscription_start_date"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <p
              className="MuiFormHelperText-root Mui-error"
              id="inscription_start_date-helper-text"
            >
              Select a valid inscription start date
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-describedby="inscription_End_date-helper-text"
                aria-invalid={true}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                disabled={false}
                id="inscription_End_date"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <p
              className="MuiFormHelperText-root Mui-error"
              id="inscription_End_date-helper-text"
            >
              Select a valid inscription end date
            </p>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="number_of_lanes"
              id="number_of_lanes-label"
            >
              Number of Lanes
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="number_of_lanes-label number_of_lanes"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                id="number_of_lanes"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                5
              </div>
              <input
                aria-describedby="number_of_lanes-helper-text"
                autoFocus={false}
                onAnimationStart={[Function]}
                type="hidden"
                value={5}
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              className="MuiFormHelperText-root MuiFormHelperText-filled"
              id="number_of_lanes-helper-text"
            >
              Please select the pool number of lanes
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <div
        className="MuiGrid-root makeStyles-actionButtons-5 MuiGrid-container MuiGrid-spacing-xs-4"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Create New Competition
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Clean All Fields
            </span>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Go Back
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
